Tests/Marketplace/upload_packs.py

```diff
@@ -158,10 +159,28 @@ def update_index_folder(index_folder_path, pack_name, pack_path):
         index_folder_subdirectories = [d for d in os.listdir(index_folder_path) if
                                        os.path.isdir(os.path.join(index_folder_path, d))]
         index_pack_path = os.path.join(index_folder_path, pack_name)
+        metadata_files_in_index = glob.glob(f"{index_pack_path}/**/metadata-*.json")
+        new_metadata_path = ''
 
+        if pack_version:
+            new_metadata_path = os.path.join(index_pack_path, f"metadata-{pack_version}.json")
+
+            # Update the latest metadata
+            if new_metadata_path in metadata_files_in_index:
+                metadata_files_in_index.remove(new_metadata_path)
+
+        # Remove old files but keep metadata files
         if pack_name in index_folder_subdirectories:
-            shutil.rmtree(index_pack_path)
-        shutil.copytree(pack_path, index_pack_path)
+            for d in os.scandir(index_pack_path):
+                if d.path not in metadata_files_in_index:
+                    os.remove(d.path)
+
+        # Copy new files and add metadata for latest version
+        for d in os.scandir(pack_path):
+            shutil.copy(d.path, index_pack_path)
+            if pack_version and Pack.METADATA in d.name:
```

Comment:
```suggestion
            if pack_version and Pack.METADATA == d.name:
```
Mainly for readability.